components/core/Taskfile.yml

```diff
@@ -0,0 +1,44 @@
+version: "3"
+
+vars:
+  BUILD_DIR: "{{.TASKFILE_DIR}}/build"
+  LINT_VENV_DIR: "{{.BUILD_DIR}}/venv"
+  PYTHON_VERSION:
+    sh: "python3 -c \"import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')\""
+
+tasks:
+  clean:
+    cmds:
+      - "rm -rf {{.BUILD_DIR}}"
+
+  lint:
+    deps: [ "lint_venv" ]
+    dir: "{{.TASKFILE_DIR}}"
+    cmds:
+      - |
+        . "{{.LINT_VENV_DIR}}/bin/activate"
+        find src tests \
+          -type f \
+          \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" -o -iname "*.inc" \) \
+          -print0 | \
+            xargs -0 clang-format --dry-run -Werror
+    sources:
+      - "src/{**/*.cpp,**/*.h,**/*.hpp,**/*.inc}"
+      - "tests/{**/*.cpp,**/*.h,**/*.hpp,**/*.inc}"
+
+  lint_venv:
```

Comment:
Nit: can we reorder them so that `lint_venv` comes before `lint` as it's a dependency

Reply:
I was following our guideline of public before private which in this case translates to parent before child; the idea being if you were reading this file from top to bottom, you'd care about reading the `lint` task before you care about reading its dependencies.